### PR TITLE
Move release build definition into PowerShell

### PR DESCRIPTION
**Pull request by TravisEz13**
1.  Use Docker Based Build module to create a build definition
    1. Create docker files that define images capable of building PowerShell
    1. Create the build json used by Docker Based Build to define builds

Prerequisites:

1. Docker (on Windows, Docker must be set to run Windows containers)
1. Windows builds must be run on Windows
1. Linux builds must be run on Linux or macOS

Usage:
```powershell
./tools/releaseBuild/vstsbuild.ps1 -ReleaseTag v6.0.0-test.9 -Name centos.7 
```

### CODE REVIEW

File: tools/releaseBuild/vstsbuild.ps1 [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+$gitBinFullPath = (Get-Command -Name git).Source
+if (-not $gitBinFullPath)
+{
+    throw "Git is required to proceed. Install from 'https://git-scm.com/download/win'"
+}
+
+Write-Verbose "cloning -b $psReleaseBranch --quiet https://github.com/$psReleaseFork/PSRelease.git" -verbose
+& $gitBinFullPath clone -b $psReleaseBranch --quiet https://github.com/$psReleaseFork/PSRelease.git $location
```
Comment (daxian-dbw) [inline tools/releaseBuild/vstsbuild.ps1]: Why not move those foundation modules to powershell repo? Wouldn't it be clear/easier to find everything in one place?

**Review by daxian-dbw — DISMISSED** [review]

File: tools/releaseBuild/vstsbuild.ps1 (hunk 1 above)
Comment (TravisEz13) [reply tools/releaseBuild/vstsbuild.ps1]: I need to deliver this to other teams.  PowerShell shouldn't be the one big monolithic project that owns very module.

Comment (daxian-dbw) [reply tools/releaseBuild/vstsbuild.ps1]: OK, so we want those modules to be reused by other teams. That makes sense.

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1 [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+$releaseTagParam = @{}
+if($ReleaseTag)
+{
+    $releaseTagParam = @{ 'ReleaseTag' = $ReleaseTag }
+}
+
+if(-not $env:homedrive)
+{
+    Write-Verbose "fixng empty home paths..." -Verbose
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1]: Typo in fixing

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1 [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    $profileParts = $env:userprofile -split ':'
+    $env:homedrive = $profileParts[0]+':'
+    $env:homepath = $profileParts[1]
+}
+
+if(! (Test-Path $destination))
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1]: If you are using -Force for New-Item then you do not need to Test-Path.

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1 [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    Start-PSPackage @pspackageParams @releaseTagParam
+
+    Write-Verbose "Exporting packages ..." -verbose
+    $files= @()
+
+    Get-ChildItem $location\*.msi |Select-Object -ExpandProperty FullName | ForEach-Object {
+        $files += $_
+    }
+
+    Get-ChildItem $location\*.zip |Select-Object -ExpandProperty FullName | ForEach-Object {
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1]: All of the below can be combined to:

```Powershell
Get-ChildItem $location\*.msi,$location\*.zip |  Copy-Item -Destination "$destination\" -Force 
```

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1 (hunk 4 above)
Comment (TravisEz13) [reply tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1]: I simplified, but I want to keep the write verbose...

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1 (hunk 3 above)
Comment (TravisEz13) [reply tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1]: Removed the force...

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1 (hunk 2 above)
Comment (TravisEz13) [reply tools/releaseBuild/Images/microsoft_powershell_windowsservercore/PowerShellPackage.ps1]: fixed

**Comment by TravisEz13** [comment]
@adityapatwardhan *ping*

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Comment by adityapatwardhan** [comment]
@TravisEz13 Have the docker files been tested?

**Comment by TravisEz13** [comment]
The Docker files have been tested along with a new bulid definition

**Pull request merged**